include/swift/ABI/GenericContext.h

```diff
@@ -467,7 +527,84 @@ class TrailingGenericContextObjects<TargetSelf<Runtime>,
   }
-  
+
+  bool hasConditionalSuppressedProtocols() const {
+    if (!asSelf()->isGeneric())
+      return false;
+
+    return getGenericContextHeader().hasConditionalSuppressedProtocols();
+  }
+
+  const SuppressibleProtocolSet &
+  getConditionalSuppressedProtocols() const {
+    assert(hasConditionalSuppressedProtocols());
+    return *this->template
+        getTrailingObjects<ConditionalSuppressibleProtocolSet>();
+  }
+
+  /// Retrieve the counts for # of conditional suppressible protocols for each
+  /// conditional conformance to a suppressible protocol.
+  ///
+  /// The counts are cumulative, so the first entry in the array is the
+  /// number of requirements for the first conditional conformance. The
+  /// second entry in the array is the number of requirements in the first
+  /// and second conditional conformances. The last entry is, therefore, the
+  /// total count of requirements in the structure.
+  llvm::ArrayRef<ConditionalSuppressibleProtocolsRequirementCount>
+  getConditionalSuppressibleProtocolRequirementCounts() const {
+    if (!asSelf()->hasConditionalSuppressedProtocols())
+      return {};
+
+    return {
+      this->template
+        getTrailingObjects<ConditionalSuppressibleProtocolsRequirementCount>(),
+      getNumConditionalSuppressibleProtocolsRequirementCounts()
+    };
+  }
+
+  /// Retrieve the array of requirements for conditional conformances to
+  /// the ith conditional conformance to a suppressible protocol.
+  llvm::ArrayRef<GenericConditionalSuppressibleProtocolRequirement>
+  getConditionalSuppressibleProtocolRequirementsAt(unsigned i) const {
+    auto counts = getConditionalSuppressibleProtocolRequirementCounts();
+    assert(i < counts.size());
+
+    unsigned startIndex = (i == 0) ? 0 : counts[i-1].count;
+    unsigned endIndex = counts[i].count;
+
+    auto basePtr =
+      this->template
+        getTrailingObjects<GenericConditionalSuppressibleProtocolRequirement>();
+    return { basePtr + startIndex, basePtr + endIndex };
+  }
+
+  /// Retrieve the array of requirements for conditional conformances to
+  /// the ith conditional conformance to a suppressible protocol.
+  llvm::ArrayRef<GenericConditionalSuppressibleProtocolRequirement>
+  getConditionalSuppressibleProtocolRequirementsFor(
+      SuppressibleProtocolKind kind
+  ) const {
+    if (!asSelf()->hasConditionalSuppressedProtocols())
+      return { };
+
+    auto conditionallySuppressed = getConditionalSuppressedProtocols();
+    if (!conditionallySuppressed.contains(kind))
+      return { };
+
+    // Count the number of "set" bits up to (but not including) the
+    // bit we're looking at.
+    unsigned targetBit = static_cast<uint8_t>(kind);
+    auto suppressedBits = conditionallySuppressed.rawBits();
+    unsigned priorBits = 0;
+    for (unsigned i = 0; i != targetBit; ++i) {
+      if (suppressedBits & 0x01)
+        ++priorBits;
+      suppressedBits = suppressedBits >> 1;
+    }
```

Comment:
We'd probably need to pull in the header. I'd like to handle this as a follow-up, because dragging another LLVM header into the runtime is the kind of thing that will find a way to blow up on me somewhere.